### ISSUE TITLE
Interoperable discovery with XTypes (PID 0x75)

### ISSUE THREAD

**Issue by mitza-oci** [eProsima/Fast-DDS#1762]
Fast-DDS doesn't parse discovery announcements from peers using XTypes (PID 0x75)

## Expected Behavior
Since Fast-DDS endpoints (at least the ones seen in the Shapes Demo) aren't announced with XTypes (PID 0x75), this PID could be ignored when received from a non-Fast-DDS endpoint.

## Current Behavior
The parsing code ends up at eprosima::fastcdr::exception::NotEnoughMemoryException and rejects the message.

## Steps to Reproduce
1. Start Fast-DDS Shapes Demo, create a writer on a topic
2. Start [OpenDDS Shapes Demo](https://opendds.org/quickstart/GettingStartedShapesDemo.html)
3. In OpenDDS's demo, create a reader on the same topic
4. Fast-DDS doesn't match this reader.  GDB shows an exception thrown within parsing when trying to create a vector of unreasonable size.

## System information

- Fast-RTPS version: 70e2ad09001cbe92eea7bfe66830099b96b6f7d4
- OS: Ubuntu 18.04

## Additional context
The contents of PID 0x75 are specified as:
```
// The types in this file shall be serialized with XCDR encoding version 2
@extensibility(MUTABLE) @nested
struct TypeInformation {...
```

**Comment by MiguelCompany** [comment]
@mitza-oci We have already identified this issue. PR #1752 is doing what you propose (ignoring the type information PID when the serialization format is not the one expected by Fast DDS).

**Comment by mitza-oci** [comment]
> 
> 
> @mitza-oci We have already identified this issue. PR #1752 is doing what you propose (ignoring the type information PID when the serialization format is not the one expected by Fast DDS).

Thanks for the info, I didn't realize that PR was related.

**Comment by adamsj-oci** [comment]
I also think this change would be good on version 2.0.0 which I believe is the version for ROS2 Foxy.

**Comment by MiguelCompany** [comment]
Yeah. We will backport #1752 after it is merged into master

**Comment by MiguelCompany** [comment]
@mitza-oci #1752 has been merged and is also on 2.1.x and 2.0.x, so I think this can be closed.

**Comment by MiguelCompany** [comment]
Closing as the fix is already present on all 2.x.x branches
